### ISSUE TITLE
[all] cloudbuild failing

### ISSUE THREAD

**Issue by mdbooth** [kubernetes/cloud-provider-openstack#2136]
Example: https://prow.k8s.io/view/gs/kubernetes-jenkins/logs/cloud-provider-openstack-push-images/1630385626224267264

The immediate cause seems to be an incorrect go version, but as the build method looks highly unusual it is probably simpler to move to a more orthodox build method. Specifically, the docker builds should be self-contained. They currently build a rootfs.tar from the Makefile and copy it into an alpine image.

I suggest copying the structure of a Dockerfile from another project would be simplest. We will likely have to update cloudbuild.yaml afterwards as well.

This impacts #1540

**Comment by mdbooth** [comment]
Should find a list of image pushes here: https://testgrid.k8s.io/provider-openstack-openstack-cloud-controller-manager#cloud-provider-openstack-image-pushes

**Comment by jichenjc** [comment]
I assume you are talking about something like following sample?
https://github.com/sustainable-computing-io/kepler/blob/main/build/Dockerfile
like from a base and we build from the base image (to keep the build env stable) ?

anyway, should we consider fix this first issue first to unblock then open another issue to track the build env things? 
or you want to fix the build method first ?

**Comment by mdbooth** [comment]
> I assume you are talking about something like following sample? https://github.com/sustainable-computing-io/kepler/blob/main/build/Dockerfile like from a base and we build from the base image (to keep the build env stable) ?
> 
> anyway, should we consider fix this first issue first to unblock then open another issue to track the build env things? or you want to fix the build method first ?

It was going to get messy to fix because we were attempting to build in the test-infra image. I don't think it was intended to be used that way, so they weren't likely to be sympathetic to change requests, and creating our own cloudbuild image would come with its own issues.
